python/rtypes/pcc/triggers.py

```diff
@@ -0,0 +1,49 @@
+
+
+class TriggerTime(object):
+    before = "b"
+    after = "a"
+
+
+class TriggerAction(object):
+    create = "c"
+    read = "r"
+    update = "u"
+    delete = "d"
+
+
+class TriggerProcedure(object):
+    def __init__(self, procedure, pcc_type, time, action, priority):
+        self.procedure = procedure
+        self.pcc_type = pcc_type
+        self.time = time
+        self.action = action
+        self.priority = priority 
+
+    def __call__(self, dataframe=None, new=None, old=None, current=None):
+        return self.procedure(dataframe, new, old, current)
+
+    def __eq__(self, v):
+        return v.priority == self.priority
```

Comment:
add pcc_type, and time and action to the equality too. It doesnt make sense that triggers for different types and same priority are considered the same. This should also fix your problem in __procedure_index_location in trigger_manager.py